### PR TITLE
feat(jellyfish-api-core): add deriveAddresses function

### PR DESCRIPTION
**Pull request by DocteurPing**
#### What this PR does / why we need it:

This PR adds the function deriveAddresses.

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should derive an address without range', async () => {
+    const descriptor = 'wpkh(tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/1/1/0)#t6wfjs64'
+    const address = ['bcrt1qjqmxmkpmxt80xz4y3746zgt0q3u3ferr34acd5']
+    expect(await client.misc.deriveAddresses(descriptor)).toStrictEqual(address)
```
Comment (shohamc1) [inline packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts]: You should also check for the exact error thrown.

```suggestion
    const promise = client.misc.deriveAddresses(descriptor)).toStrictEqual(address)
    await expect(promise).rejects.toThrow(RpcApiError)
    await expect(promise).rejects.toMatchObject({
      payload: {
        code: -5,
        message: '<err message>',
        method: 'deriveaddresses'
      }
    })
```

File: packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const descriptor = 'wpkh(tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/1/1/*)'
+    await expect(client.misc.deriveAddresses(descriptor, [100000])).rejects.toThrow(RpcApiError)
+  })
+
+  it('should raise an error if range is too large', async () => {
+    const descriptor = 'wpkh(tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/1/1/*)'
+    await expect(client.misc.deriveAddresses(descriptor, [1, 100000])).rejects.toThrow(RpcApiError)
+  })
+
+  it('should raise an error if range is inferior to 0', async () => {
```
Comment (shohamc1) [inline packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts]:
```suggestion
  it('should raise an error if range is less than 0', async () => {
```

File: packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    const descriptor = 'wpkh(tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/1/1/*)'
+    await expect(client.misc.deriveAddresses(descriptor, [-1, 2])).rejects.toThrow(RpcApiError)
+  })
+
+  it('should raise an error if end of range is inferior to beginning', async () => {
```
Comment (shohamc1) [inline packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts]:
```suggestion
  it('should raise an error if end of range is smaller than the start', async () => {
```

**Review by shohamc1 — CHANGES_REQUESTED** [review]
It would be worth testing other output descriptors like `pkh` and `sh` which are also supported by the RPC.

File: packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  it('should derive an address without range', async () => {
+    const descriptor = 'wpkh(tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/1/1/0)#t6wfjs64'
+    const address = ['bcrt1qjqmxmkpmxt80xz4y3746zgt0q3u3ferr34acd5']
+    expect(await client.misc.deriveAddresses(descriptor)).toStrictEqual(address)
+  })
+
+  it('should derive an address with range', async () => {
+    const descriptor = 'wpkh(tprv8ZgxMBicQKsPd7Uf69XL1XwhmjHopUGep8GuEiJDZmbQz6o58LninorQAfcKZWARbtRtfnLcJ5MQ2AtHcQJCCRUcMRvmDUjyEmNUWwx8UbK/1/1/*)#kft60nuy'
+    const address = ['bcrt1qhku5rq7jz8ulufe2y6fkcpnlvpsta7rq4442dy', 'bcrt1qpgptk2gvshyl0s9lqshsmx932l9ccsv265tvaq']
+    expect(await client.misc.deriveAddresses(descriptor, [1, 2])).toStrictEqual(address)
```
Comment (shohamc1) [inline packages/jellyfish-api-core/__tests__/category/misc/deriveAddresses.test.ts]:
```suggestion
    await expect(client.misc.deriveAddresses(descriptor, [1, 2])).toStrictEqual(address)
```

**Review by shohamc1 — CHANGES_REQUESTED** [review]

**Review by shohamc1 — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**